src/codeu/chat/common/Message.java

```diff
@@ -55,7 +57,8 @@ public Message read(InputStream in) throws IOException {
   };
 
   public final Uuid id;
-  public final Uuid previous;
+  //TODO: got rid of final privacy setting, unsure if this is the right solution
+  public Uuid previous;
```

Comment:
I got rid of "final" because when I delete messages, I update the "next" and "previous" fields of the Message before and the Message after the Message to be deleted. "next" wasn't "final", so I had no problems updating that, but "previous" was. I'll ask Elmer about this during our meeting.